### ISSUE TITLE
support for keyboard hints like vimium/vimperator/surfingkeys/tmux-{fingers,thumbs,fastcopy,picker}

### ISSUE THREAD

**Issue by aspiers** [woodstok/tmux-butler#10]
Hi, thanks for this great project!

I've just done a quick comparison of the following tpm plugins:

- tmux-butler
- tmux-fingers
- tmux-thumbs
- tmux-fastcopy
- tmux-picker
- tmux-copycat
- extrakto

They're all great, with a lot of overlap in their essential goal of quick copy/paste/actions on tmux buffer text.  Admittedly I've not looked too deep into the code of each, but to be honest it looks like tmux-butler has the nicest architecture: it's easy to understand, modular, and highly extensible.  The modes feature in particular is really awesome.

In fact, the only thing stopping me from using it is that there is no native support in the Selector stage for anything other than fzf.  I know that the selector can be set to anything, but for me, using the keyboard hints style of selection used by vimium / vimperator / surfingkeys / tmux-{fingers,thumbs,fastcopy,picker} and others can be quicker than fzf style in many cases.  That said, I would love a single tool which supports both fzf and hints selection.

So do you have any suggestions on how hints selection could be best implemented within tmux-butler?

Thanks again!

**Comment by woodstok** [comment]
Hmm. Interesting. I was myself looking at an alternative vimium style picker because there are times when the fzf popup would take away the focus/context on what I wanted to quickly grab through butler. 
But, having a picker can also have limitations when broken up tokens/strings are overlapping. Ill put some thought into it this weekend.

**Comment by aspiers** [comment]
Yes, the context hiding is certainly one of the downsides of the fzf popup.  Another is that it's not always convenient to type a substring, e.g. if the desired string is something like `0x40188c8be97c8ae69851350f15dcd15c61359e93e8100b6f3f4ed2c13491e53e`.

However, of course there are advantages to the fzf popup approach too.  My solution is to use both: I previously set up `C-o` to trigger `tmux-fingers` and `C-M-o` to trigger `extrakto`, but I'd much prefer to use `tmux-butler` for both due to its superior flexibility.  Perhaps we can steal the picker code from `tmux-fingers` or `tmux-thumbs` or `tmux-picker` or `tmux-fastcopy`.
